test/stubadub/core_test.cljc

```diff
@@ -1,36 +1,100 @@
 (ns stubadub.core-test
-  #?(:clj  (:require [clojure.test :refer [deftest is]]
-                     [stubadub.core :refer [with-stub calls-to]])
-     :cljs (:require [cljs.test :refer-macros [deftest is]]
-                     [stubadub.core :refer [calls-to] :refer-macros [with-stub]])))
+  #?(:clj  (:require [clojure.test :refer [deftest is testing]]
+                     [stubadub.core :as sut])
+     :cljs (:require [cljs.test :refer-macros [deftest is testing]]
+                     [stubadub.core :as sut :include-macros true])))
 
 #?(:cljs (defn spit [& args])) ;; these don't exist when running cljs-tests on node.js
 #?(:cljs (defn slurp [& args]))
 
-(deftest calls-to-returns-a-list-of-the-arguments-passed-to-the-stub
+(deftest stub
+  (testing "a stub is a function"
+    (is (ifn? (sut/stub))))
+
+  (testing "a stub returns nil by default"
+    (let [stub (sut/stub)]
+      (is (nil? (stub)))))
+
+  (testing "a stub can take any number of arguments"
+    (let [stub (sut/stub)]
+      (is (nil? (stub 1)))
+      (is (nil? (stub 1 2)))
+      (is (nil? (stub 1 2 3)))
+      ;; more than 20 args to stub not supported on cljs
```

Comment:
Is this fixed by using `& more` in the Stub definition?

Reply:
I'm afraid not. `clojure.lang.IFn` defined `applyTo` to cover invocations with arguments in excess of 20. Clojurescript's equivalent `cljs.core/IFn` does not. I was informed by David Nolen via Slack that this simply was not implemented in cljs (yet). To be precise, this limitation is for function defined via `IFn`, not functions defined via `fn`.

Interresting side note; Clojure does not allow you to define a function (via `fn`) with more than 20 required positional arguments; the rest must go as `& more`. Clojurescript, however, does not have this limit.

Not being able to stub any function allowed by the language sucks (and is indeed a breaking change to stubadub), although I can hardly imaging this being a practical problem (famous last words). 

I'll take a stab at implementing this in compiler later this month.